drivers/timer/esp32c3_sys_timer.c

```diff
@@ -32,7 +32,14 @@ static uint64_t last_count;
 
 static void set_systimer_alarm(uint64_t time)
 {
+
+#if defined(CONFIG_TICKLESS_KERNEL)
```

Comment:
This seems backwards.  When TICKLESS, we don't want the timer to be retriggered, we want it to fire exactly once, at the time requested via sys_clock_set_timeout().

Reply:
@andyross sure, I removed this commit. Just added the missing IRQ value in test case.